src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -1421,7 +1421,7 @@ private String getFullPath(final String featureString) {
         private String dirname(String path) {
             final int lastSlash = path.lastIndexOf(File.separatorChar);
             if (lastSlash == -1) {
-                return path;
+                return getContext().send(getContext().getCoreLibrary().getDirClass(), "getwd").toString();
```

Comment:
Use `FeatureLoader.getWorkingDirectory()` (needs to be made public)

Reply:
What if the passed path contains a `/` but is still relative (like `a/b`)?
I think it should still be interpreted relative to CWD in such a case.

Reply:
I'll add a spec and find out what happens.